src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1016,26 +1015,29 @@ def _get_auth_provider_latest_api_version(cli_ctx):
 def _update_provider(cli_ctx, namespace, registering, wait, properties=None, mg_id=None, accept_terms=None):
     import time
     target_state = 'Registered' if registering else 'Unregistered'
-    client = cf_providers(cli_ctx, None, api_version='2021-04-01')
+    rcf = _resource_client_factory(cli_ctx)
     is_rpaas = namespace.lower() in RPAAS_APIS
     if mg_id is None and registering:
         if is_rpaas and accept_terms:
             wait = True
-        r = client.register(namespace, properties=properties)
+        if properties:
+            r = rcf.providers.register(namespace, properties=properties)
+        else:
+            r = rcf.providers.register(namespace)
```

Comment:
I'd like to ask why modify this logic? When the `properties` is `None`, the parameters is not allowed to be passed in as `rcf.providers.register(namespace, properties=None)`, right?

Reply:
When using old version, like profile hybird 2018-03-01, function register doesn't have parameter `properties`. It will cause an Error. So if properties is None, we won't set properties in register.

Reply:
Because this logic is used to be compatible with the old version of SDK, so it may be more appropriate to use the api-version to distinguish this logic